tracer/test/Datadog.Trace.Tests/TraceContextTests.cs

```diff
@@ -195,16 +195,70 @@ public void PartialFlushShouldPropagateMetadata(bool inAASContext)
             tracer.Setup(t => t.Write(It.IsAny<ArraySegment<Span>>()))
                   .Callback<ArraySegment<Span>>(s => spans = s);
 
-            var vars = Environment.GetEnvironmentVariables();
+            SetAASContext(inAASContext);
+            var traceContext = new TraceContext(tracer.Object);
+            traceContext.SetSamplingPriority(SamplingPriorityValues.UserKeep);
 
-            if (inAASContext)
+            var rootSpan = CreateSpan();
+
+            traceContext.AddSpan(rootSpan);
+
+            for (int i = 0; i < partialFlushThreshold - 1; i++)
             {
-                vars.Add(AzureAppServices.AzureAppServicesContextKey, "true");
-                vars.Add(AzureAppServices.ResourceGroupKey, "ThisIsAResourceGroup");
-                vars.Add(Datadog.Trace.Configuration.ConfigurationKeys.ApiKey, "xxx");
-                AzureAppServices.Metadata = new AzureAppServices(vars);
+                var span = CreateSpan();
+                traceContext.AddSpan(span);
+                traceContext.CloseSpan(span);
             }
 
+            // At this point, only one span is missing to reach the threshold for partial flush
+            spans.Should().BeNull("partial flush should not have been triggered");
```

Comment:
Since `PartialFlushShouldPropagateMetadata` is already testing the partial flush scenario, I suggest not using partial flush in this test. The test will look similar because you'll close all of the child spans and the root span, and then after the root span is closed you'll open and close one additional span.

Reply:
done in  ee9cc86cd